config/notifiers.go

```diff
@@ -156,6 +165,24 @@ func (nc *NotifierConfig) SendResolved() bool {
 	return nc.VSendResolved
 }
 
+// DiscordConfig configures notifications via Discord.
+type DiscordConfig struct {
+	NotifierConfig `yaml:",inline" json:",inline"`
+
+	HTTPConfig *commoncfg.HTTPClientConfig `yaml:"http_config,omitempty" json:"http_config,omitempty"`
+	APIURL     *SecretURL                  `yaml:"api_url,omitempty" json:"api_url,omitempty"`
+
+	Title   string `yaml:"title,omitempty" json:"title,omitempty"`
+	Message string `yaml:"message,omitempty" json:"message,omitempty"`
```

Comment:
I know this might be a bit more work, but I'd love it if we can add two new options:

- Avatar URL `avatar_url`
- Username `username`

(they belong on the main webhook object and not the embedded, see: https://discord.com/developers/docs/resources/webhook#execute-webhook)

For the sake of transparency, Grafana supports both of these options in its discord notifier and I'd love it if they could be uniform. Being conscious that we're all short on time, I'm happy to address my own comments on top of this PR in case you don't have the time to execute them.

Reply:
One thing about alertmanager-discord is that the username is for some use cases is set to "firing/resolved" because on push notifications, embeds don't show up as easily, so it's easier to work with at a glance

Reply:
I don't really have a strong opinion on this. Would the two of you be fine with taking this into an issue to follow up on and then decide with how to move forward outside this PR?